pkg/tls/self_signed_cert_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package gencert generates self-signed TLS certificates.
+package tls
+
+import (
+	"crypto/tls"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateSelfSignedCert(t *testing.T) {
+	// Setup temp file paths
+	certPath := "test_cert.pem"
+	keyPath := "test_key.pem"
+
+	// Cleanup any pre-existing files from previous tests
+	defer os.Remove(certPath)
+	defer os.Remove(keyPath)
+
+	// Test case 1: Cert file already exists
+	t.Run("Test 1: Cert file exists", func(t *testing.T) {
```

Comment:
Could a struct be used for each test case instead of having multiple `t.Run` with the same checks ? 
Example
https://github.com/nginx/agent/blob/f5a420fa44d7fd7ea52fba962539cf681fe40b38/pkg/files/file_helpers_test.go#L57-L61

Reply:
Done!